packages/dataviews/src/types/field-api.ts

```diff
@@ -92,6 +92,80 @@ export type Rules< Item > = {
 		  ) => Promise< null | string > );
 };
 
+export type NormalizedRules< Item > = {
+	required:
```

Comment:
We should definitely make this generic and not hard coded the constraints. Folks will be able to add their own custom constraints.